docs/pages/auth-concept/authentication.md

```diff
@@ -0,0 +1,181 @@
+
+- userPassword
+- Bearer token
+- HttpPApiKey
+- ApiKey
+
+A sample `asyncapi.yaml` for a server with security requirements and a `userPassword` security schemes is shown below:
+
+```yaml
+##server asyncAPI schema
+asyncapi: 2.6.0
+info:
+  title: AsyncAPI IMDB server
+  version: 1.0.0
+  description: This app is a dummy server that would stream the trending/upcoming anime.
+servers:
+  trendingAnimeServer:
+    url: 'http://localhost:8081'
+    protocol: http
+    security:
+      - userPass: []
+
+  ...
+
+components:
+  securitySchemes:
+    userPass:
+      type: userPassword
+
+```
+
+A sample `asyncapi.yaml` for a client that implements some of the requirements of the server above:
+
+```yaml
+##client asyncAPI schema
+servers:
+  trendingAnime:
+    url: http://localhost:8081
+    protocol: http
+    security:
+      - userPass: []
+  testwebhook:
+    url: ws://localhost:9000
+    protocol: ws
+x-remoteServers:
+  - trendingAnime
+
+  ...
+
+components:
+  securitySchemes:
+    userPass:
+      type: userPassword
+
+```
+
+**The Client asyncapi.yaml file does not need to implement all the security requirements in the server, it only needs to implement the ones that it uses.**
+
+
+Glee can act as both a server and a client at the same time. Hence the need for `serverAuth` and `clientAuth`. Glee acts as a client when the `x-remoteServers` property is present in the `asyncapi.yaml` file.
+
+When Glee acts as a client, it can connect to a Glee server, and when Glee acts as a server it accepts connection from other Glee clients. Hence a Glee application can both accept connections from clients while also sending requests to other Glee applications (servers) ath the same time.
+
+When a security requirement is specified in the `asyncapi.yaml` file and Glee acts as a server, the `serverAuth` function should be implemented, if Glee acts as a client then the `clientAuth` function should be implemented. If Glee is being used as both client and server, then it should have both the `clientAuth` and `serverAuth` functions.
+
+## Server Authentication in Glee
+
+The `serverAuth` function takes an argument that can be destructured as follows
+
+| Attribute  | Description                                                     |
+| ---------- | --------------------------------------------------------------- |
+| done       | The done function that tells the server to proceed.             |
+| authProps  | The authentication parameters recieved from the client.         |
+| serverName | The name of the server/broker from which the event was emitted. |
+| doc        | The parsedAsyncAPI schema                                       |
+
+### done()
+
+The `done` parameter in the `serverAuth` function allows the broker/server to know what to do next depending on the boolean value you pass to it.
+
+```js
+/* websocket.js */
+
+export async function serverAuth({ authProps, done }) {
+  // done(true)
+  //done(false, 401, "Unauthorized")
+  // done(false)
```

Comment:
done can accept three parameters. please document this.

Reply:
sorry @KhudaDad414 Are you saying abt this `done(false, 401, "Unauthorized")`?
```
to know what to do next depending on the boolean value you pass to it.
```
For this, I think I can modify the abv line to
```
to know what to do next depending on the boolean value you pass to it along with status code and access type.
```

Reply:
I mean the `done()` function parameters should be explained as well. something like:

#### Parameters
1. **Authentication Result (`Boolean`)**: `true` for success, `false` for failure.
2. **HTTP Status Code (`Integer`)**: Code for authentication failure (e.g., 401 for Unauthorized).
3. **Status Message (`String`)**: Description of the authentication result (e.g., "Unauthorized").